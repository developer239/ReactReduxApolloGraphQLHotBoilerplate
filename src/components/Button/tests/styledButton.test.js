import React from 'react'
import { shallow } from 'enzyme'
import { theme } from 'styles'
import 'jest-styled-components'
import { StyledButton } from '../index'


describe('StyledButton Component', () => {
  it('has correct default styles', () => {
    const wrapper = shallow(
      <StyledButton theme={theme}>Button Text</StyledButton>,
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('has correct :active background-color for bgType', () => {
    const wrapper = shallow(
      <StyledButton
        bgType="success"
        theme={theme}
      >
        Button Text
      </StyledButton>,
    )
    expect(wrapper).toMatchSnapshot()
  })
})
