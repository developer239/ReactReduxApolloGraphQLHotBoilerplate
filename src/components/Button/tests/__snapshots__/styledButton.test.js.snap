// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledButton Component has correct :active background-color for bgType 1`] = `
.c0 {
  background-color: #48cfad;
  border: none;
  color: white;
  padding: 12px 20px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  font-size: 14px;
  cursor: pointer;
  -webkit-transition: background-color 150ms linear;
  transition: background-color 150ms linear;
  margin-right: 2px;
}

.c0:hover {
  background-color: #37bc9b;
}

.c0:focus {
  outline: 0;
}

.c0:active {
  background-color: #ccd1d9;
}

<button
  className="c0"
>
  Button Text
</button>
`;

exports[`StyledButton Component has correct default styles 1`] = `
.c0 {
  background-color: #ccd1d9;
  border: none;
  color: white;
  padding: 12px 20px;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  font-size: 14px;
  cursor: pointer;
  -webkit-transition: background-color 150ms linear;
  transition: background-color 150ms linear;
  margin-right: 2px;
}

.c0:hover {
  background-color: #aab2bd;
}

.c0:focus {
  outline: 0;
}

.c0:active {
  background-color: #5d9cec;
}

<button
  className="c0"
>
  Button Text
</button>
`;
